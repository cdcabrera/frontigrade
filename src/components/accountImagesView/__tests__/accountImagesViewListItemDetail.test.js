import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import {
  ConnectedAccountImagesViewListItemDetail,
  AccountImagesViewListItemDetail
} from '../accountImagesViewListItemDetail';

describe('AccountImagesViewListItemDetail Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component with default props', () => {
    const store = generateEmptyStore({
      accountImageEdit: {
        2: {
          image: {
            created_at: '2018-07-3',
            ec2_ami_id: 'ami-plain',
            id: 2,
            inspection_json: null,
            is_cloud_access: true,
            is_encrypted: false,
            is_marketplace: false,
            name: 'my favorite image',
            openshift: false,
            openshift_challenged: false,
            openshift_detected: false,
            owner_aws_account_id: '000000000000',
            platform: 'none',
            resourcetype: 'AwsMachineImage',
            rhel: true,
            rhel_challenged: true,
            rhel_detected: true,
            rhel_enabled_repos_found: true,
            rhel_product_certs_found: true,
            rhel_release_files_found: true,
            rhel_signed_packages_found: true,
            status: 'pending',
            updated_at: '2018-07-3',
            url: 'http://cloudigrade.127.0.0.1.nip.io/api/v1/image/2/'
          }
        }
      }
    });

    const props = {
      id: 2
    };
    const component = shallow(<ConnectedAccountImagesViewListItemDetail {...props} />, { context: { store } });

    expect(component).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {
      id: 2,
      image: {
        created_at: '2018-07-3',
        ec2_ami_id: 'ami-plain',
        id: 2,
        inspection_json: null,
        is_cloud_access: true,
        is_encrypted: false,
        is_marketplace: false,
        name: 'my favorite image',
        openshift: false,
        openshift_challenged: false,
        openshift_detected: false,
        owner_aws_account_id: '000000000000',
        platform: 'none',
        resourcetype: 'AwsMachineImage',
        rhel: true,
        rhel_challenged: true,
        rhel_detected: true,
        rhel_enabled_repos_found: true,
        rhel_product_certs_found: true,
        rhel_release_files_found: true,
        rhel_signed_packages_found: true,
        status: 'pending',
        updated_at: '2018-07-3',
        url: 'http://cloudigrade.127.0.0.1.nip.io/api/v1/image/2/'
      }
    };

    const component = mount(<AccountImagesViewListItemDetail {...props} />);

    expect(component.render()).toMatchSnapshot('non-connected');
  });
});
