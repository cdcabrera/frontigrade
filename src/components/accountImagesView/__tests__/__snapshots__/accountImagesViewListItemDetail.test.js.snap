// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountImagesViewListItemDetail Component should render a connected component with default props: connected 1`] = `
<Translate(AccountImagesViewListItemDetail)
  getAccountImage={[Function]}
  id={2}
  image={
    Object {
      "created_at": "2018-07-3",
      "ec2_ami_id": "ami-plain",
      "id": 2,
      "inspection_json": null,
      "is_cloud_access": true,
      "is_encrypted": false,
      "is_marketplace": false,
      "name": "my favorite image",
      "openshift": false,
      "openshift_challenged": false,
      "openshift_detected": false,
      "owner_aws_account_id": "000000000000",
      "platform": "none",
      "resourcetype": "AwsMachineImage",
      "rhel": true,
      "rhel_challenged": true,
      "rhel_detected": true,
      "rhel_enabled_repos_found": true,
      "rhel_product_certs_found": true,
      "rhel_release_files_found": true,
      "rhel_signed_packages_found": true,
      "status": "pending",
      "updated_at": "2018-07-3",
      "url": "http://cloudigrade.127.0.0.1.nip.io/api/v1/image/2/",
    }
  }
  updateAccountImageFieldRhel={[Function]}
  updateAccountImageFieldRhocp={[Function]}
/>
`;

exports[`AccountImagesViewListItemDetail Component should render a non-connected component: non-connected 1`] = `
<div
  class="fadein container-fluid"
>
  <div
    class="row"
  >
    <div
      class="col-xs-6"
    >
      <h5>
        <strong>
          Red Hat Enterprise Linux is detected
        </strong>
      </h5>
      <ul
        class="cloudmeter-list"
      >
        <li>
          t('list-images.rhel.rhel-detail-enabled-repos')
        </li>
        <li>
          t('list-images.rhel.rhel-detail-product-certs')
        </li>
        <li>
          t('list-images.rhel.rhel-detail-release-files')
        </li>
        <li>
          t('list-images.rhel.rhel-detail-signed-packages')
        </li>
        <li>
          t('list-images.rhel.rhel-detail-cloud-access')
        </li>
      </ul>
      <p>
        t('list-images.rhel.rhel-not-running')
      </p>
      <form
        class=""
      >
        <div
          class="form-group"
        >
          <div
            class="checkbox"
          >
            <label
              title=""
            >
              <input
                checked=""
                type="checkbox"
              />
              Flagged for review 
              <span
                aria-hidden="true"
                class="fa fa-flag cloudmeter-pficon-error"
              />
            </label>
          </div>
        </div>
      </form>
    </div>
    <div
      class="col-xs-6"
    >
      <h5>
        <strong>
          Red Hat OpenShift Container Platform is not detected
        </strong>
      </h5>
      <p>
        t('list-images.rhocp.rhocp-running')
      </p>
      <form
        class=""
      >
        <div
          class="form-group"
        >
          <div
            class="checkbox"
          >
            <label
              title=""
            >
              <input
                type="checkbox"
              />
              Flag for review 
            </label>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
